locking/locks_test.go

```diff
@@ -0,0 +1,149 @@
+package locking
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/git-lfs/git-lfs/api"
+	"github.com/git-lfs/git-lfs/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLockCache(t *testing.T) {
+	var err error
+
+	oldStore := config.LocalGitStorageDir
+	config.LocalGitStorageDir, err = ioutil.TempDir("", "testCacheLock")
+	assert.Nil(t, err)
+	defer func() {
+		os.RemoveAll(config.LocalGitStorageDir)
+		config.LocalGitStorageDir = oldStore
+	}()
+
+	client, err := NewClient(config.NewFrom(config.Values{}))
```

Comment:
Here's another place where I think the approach I proposed above would yield a win. This test is testing the behavior of caching a lock, but creates a locking client to do so. Moving that behavior down into a more isolated part of the system would mean that this test can be closer to it's subject.